polyfill/lib/monthcode.mjs

```diff
@@ -2,24 +2,31 @@ import {
   String as StringCtor,
   RangeError as RangeErrorCtor,
   TypeError as TypeErrorCtor,
+  ArrayFrom,
   StringPrototypePadStart,
-  RegExpPrototypeExec
+  StringPrototypeSlice
 } from './primordials.mjs';
 
 import Call from 'es-abstract/2024/Call.js';
 import ToPrimitive from 'es-abstract/2024/ToPrimitive.js';
 
-import { monthCode as MONTH_CODE_REGEX } from './regex.mjs';
+const digitsForMonthNumber = ArrayFrom({ length: 100 }, (_, i) => (i < 10 ? `0${i}` : `${i}`));
 
 export function ParseMonthCode(argument) {
   const value = ToPrimitive(argument, StringCtor);
   if (typeof value !== 'string') throw new TypeErrorCtor('month code must be a string');
-  const match = Call(RegExpPrototypeExec, MONTH_CODE_REGEX, [value]);
-  if (!match) throw new RangeErrorCtor(`bad month code ${value}; must match M01-M99 or M00L-M99L`);
-  return {
-    monthNumber: +(match[1] ?? match[3] ?? match[5]),
-    isLeapMonth: (match[2] ?? match[4] ?? match[6]) === 'L'
-  };
+  const digits = Call(StringPrototypeSlice, value, [1, 3]);
+  const monthNumber = digits.length === 2 ? +digits | 0 : 0;
```

Comment:
An improvement I missed in my first pass: when `digits.length` is not 2, we can ensure failure in the first clause of the following `if` condition.
```suggestion
  const monthNumber = digits.length === 2 ? +digits | 0 : -1;
```